scripts/intermodal_access_egress_network.py

```diff
@@ -0,0 +1,221 @@
+                                                     driver='GeoJSON')
+        logging.info(f'Modal subsetting resulted in {len(df_stops)} stops to snap')
+
+    if network_snap_modes is not None:
+        logging.info('Building Spatial Tree')
+        spatial_tree = spatial.SpatialTree(n)
+
+        for snap_mode in network_snap_modes.split(','):
+            logging.info(f'Snapping mode: {snap_mode}')
+            sub_tree = spatial_tree.modal_subtree(modes={snap_mode})
+
+            closest_links = find_closest_links_by_step(
+                network_spatial_tree=sub_tree,
+                df_stops=df_stops,
+                step_size=step_size,
+                distance_threshold=distance_threshold
+            )
+
+            # TODO There are multiple links to choose from, for the time being we are not precious about which link is selected.
+            selected_links = closest_links.reset_index().groupby('index').first()
+            if len(selected_links) != len(df_stops):
+                logging.warning(f'Only {len(selected_links)} out of {len(df_stops)} stops found a link to snap to. '
+                                'Consider removing the distance threshold if you want all stops to find a nearest link.')
+
+            # Let's create some handy geojson outputs to verify our snapping
+            selected_links[['catchment', 'geometry']].to_file(
+                os.path.join(supporting_outputs, f'{snap_mode}_stop_catchments.geojson'), driver='GeoJSON')
+            # join to get link geoms
+            selected_links = selected_links.join(
+                sub_tree.links[['link_id', 'geometry']], how='left', on='link_id', lsuffix='_left', rsuffix='')
+            selected_links[['geometry']].to_file(
+                os.path.join(supporting_outputs, f'{snap_mode}_access_egress_links.geojson'), driver='GeoJSON')
+            # get number of stops in each catchment
+            catchment_value_counts = selected_links['catchment'].value_counts().to_dict()
+            with open(os.path.join(supporting_outputs, f'{snap_mode}_catchment_value_counts.json'), 'w') as outfile:
+                json.dump(sanitiser.sanitise_dictionary(catchment_value_counts), outfile)
+            logging.info(f'Number of stops in each catchment bin: {catchment_value_counts}')
+
+            # generate the data dictionaries for updating stops data
+            access_link_id_tag = f'accessLinkId_{snap_mode}'
+            accessible_tag = f'{snap_mode}Accessible'
+            distance_catchment_tag = f'{snap_mode}_distance_catchment_tag'
+
+            selected_links[access_link_id_tag] = selected_links['link_id'].apply(
+                lambda x: {'name': access_link_id_tag, 'class': 'java.lang.String', 'text': x})
+            selected_links[accessible_tag] = selected_links.apply(
+                lambda x: {'name': accessible_tag, 'class': 'java.lang.String', 'text': 'true'}, axis=1)
```

Comment:
the attribute values have caught me out a few times, in this case `true` versus `True` for example. I am happy with this convention, but think we need some form of attribute key/value summary to pass on down the pipeline to the next user. Suggest this could be a general "report" cli command, eg `genet report attributes <PATH>`